features/mink_integration/resetting_service_container_state.feature

```diff
@@ -0,0 +1,198 @@
+        """
+        When I run Behat
+        Then it should pass
+
+    Scenario: Driver's service container is not reset before a request is made, even when another scenario made a request before
+        # Regression testing https://github.com/FriendsOfBehat/SymfonyExtension/issues/149
+        Given a feature file containing:
+        """
+        Feature:
+            Scenario:
+                Given the counter service is zeroed
+                And I increment the counter
+                When I visit the page "/hello-world"
+                Then I should see "The counter value is 2" on the page
+            Scenario:
+                Given the counter service is zeroed
+                And I increment the counter
+                When I visit the page "/hello-world"
+                Then I should see "The counter value is 2" on the page
+        """
+        When I run Behat
+        Then it should pass
+
+    Scenario: Driver's service container can be inspected after a request has been made
+        Given a feature file containing:
+        """
+        Feature:
+            Scenario:
+                Given the counter service is zeroed
+                When I visit the page "/hello-world"
+                Then the counter service should return 1
+        """
+        When I run Behat
+        Then it should pass
+
+    Scenario: When multiple requests are made, the driver's service container is reset, but we can inspect "in between" states
+        Given a feature file containing:
+        """
+        Feature:
+            Scenario:
+                Given the counter service is zeroed
+                When I visit the page "/hello-world"
+                Then the counter service should return 1
+                # This will reset the driver's container, so we will see "1" again
+                When I visit the page "/hello-world"
+                Then the counter service should return 1
+        """
+        When I run Behat
+        Then it should pass
+
+    Scenario: Driver's service container can not reasonably be modified between requests
+        # This is not really a feature, but rather documenting current behavior (judge yourself).
+        # One way around it might be to (how?) disable the reboot feature on the KernelBrowser and
+        # take responsibility for resetting the driver's container yourself.
+        Given a feature file containing:
+        """
+        Feature:
+            Scenario:
+                Given the counter service is zeroed
+                And I increment the counter
+                When I visit the page "/hello-world"
+                Then I should see "The counter value is 2" on the page
+                And the counter service should return 2
+                # Now a second request is made, which will reset the container, but leaves us
+                # with no easy way of pre-setting the container once again:
+                When I increment the counter
```

Comment:
Ok, I see there is no silver bullet for this use case. Today, I use a custom context to force the reset of the container to allow me to re-pre-setting the container again.